stdlib/auth/src/main/ballerina/src/auth/utils.bal

```diff
@@ -72,41 +71,30 @@ public function extractUsernameAndPassword(string credential) returns [string, s
     }
 }
 
-# Sets the authentication-related values (scheme, auth token) to the authentication context of the invocation context.
+# Sets the authentication-related values to the invocation context.
 # ```ballerina
-# auth:setAuthenticationContext("jwt", "<credential>");
+# auth:setAuthInvocationContext("jwt", "<credential>", "<userID>", <scopes>, <claims>);
 # ```
 #
 # + scheme - Auth scheme (`JWT`, `LDAP`, `OAuth2`, `Basic`, etc.)
 # + authToken - Auth token (credential)
-public function setAuthenticationContext(string scheme, string authToken) {
-    runtime:InvocationContext invocationContext = runtime:getInvocationContext();
-    invocationContext.authenticationContext = {
-        scheme: scheme,
-        authToken: authToken
-    };
-}
-
-# Sets the authentication-related values (user ID, username, scopes, claims) to the principal of the invocation context.
-#
-# + userId - User ID of the authenticated user
-# + username - Username of the authenticated user
-# + claims - Claims of the authenticated user
-# + scopes - Authenticated user scopes
-public function setPrincipal(public string? userId = (), public string? username = (), public string[]? scopes = (),
-                             public map<any>? claims = ()) {
-    runtime:InvocationContext invocationContext = runtime:getInvocationContext();
-    if (!(userId is ()) && userId != "") {
-        invocationContext.principal.userId = userId;
+public function setAuthInvocationContext(public string? scheme = (), public string? authToken = (),
```

Comment:
Shall we rename this function as follow and move this function to 'auth/invocation_context.bal' ?
```suggestion
public function setInvocationContext(public string? scheme = (), public string? authToken = (),
```